pkg/apis/core/validation/validation.go

```diff
@@ -2315,19 +2332,26 @@ func ValidatePersistentVolumeClaimStatusUpdate(newPvc, oldPvc *core.PersistentVo
 		allErrs = append(allErrs, validateBasicResource(qty, capPath.Key(string(r)))...)
 	}
 	if validationOpts.EnableRecoverFromExpansionFailure {
```

Comment:
I am double checking with other api reviewers, but I think the existing [validation enablement](https://github.com/kubernetes/kubernetes/blob/e0746947826bc28e28eb4d8c7871cbe9ae126cf1/pkg/apis/core/validation/validation.go#L2026) is not quite right.

I think if the old field is set, we need to continue to validate. Otherwise, someone could mutate the field to something invalid while the feature gate is off.

Reply:
Confirmed we should continue validating when the old field is set.